__tests__/app.test.js

```diff
@@ -340,3 +340,55 @@ describe("GET /api/users", () => {
       });
   });
 });
+
+describe("GET /api/articles?topic=", () => {
+  test("200: returns articles when given a specified topic", () => {
+    return supertest(app)
+      .get("/api/articles?topic=mitch")
+      .expect(200)
+      .then((res) => {
+        const { articles } = res.body;
+        expect(articles.length > 0).toBe(true);
+
+        articles.forEach((article) => {
+          expect(article).toHaveProperty("topic", "mitch");
+          expect(article).toHaveProperty("article_id", expect.any(Number));
+          expect(article).toHaveProperty("title", expect.any(String));
+          expect(article).toHaveProperty("author", expect.any(String));
+          expect(article).toHaveProperty("body", expect.any(String));
+          expect(article).toHaveProperty("created_at", expect.any(String));
+          expect(article).toHaveProperty("votes", expect.any(Number));
+          expect(article).toHaveProperty("article_img_url", expect.any(String));
+        });
+      });
+  });
+  test("400: returns appropriate response when sending a get request with a topic that doesnt exist", () => {
+    return supertest(app)
+      .get("/api/articles?topic=non_existent_topic")
+      .expect(400)
+      .then((res) => {
+        const { msg } = res.body;
+        expect(msg).toBe("Topic does not exist");
+      });
+  });
+  test("200: returns all articles when the query is omitted", () => {
+    return supertest(app)
+      .get("/api/articles?topic")
+      .expect(200)
+      .then((res) => {
+        const { articles } = res.body;
+        expect(articles.length > 0).toBe(true);
+
+        articles.forEach((article) => {
+          expect(article).toHaveProperty("topic", expect.any(String));
+          expect(article).toHaveProperty("article_id", expect.any(Number));
+          expect(article).toHaveProperty("title", expect.any(String));
+          expect(article).toHaveProperty("author", expect.any(String));
+          expect(article).toHaveProperty("body", expect.any(String));
+          expect(article).toHaveProperty("created_at", expect.any(String));
+          expect(article).toHaveProperty("votes", expect.any(Number));
+          expect(article).toHaveProperty("article_img_url", expect.any(String));
+        });
+      });
+  });
+});
```

Comment:
Missing a test for `topic=paper` this is a topic with no articles in it, so this is completely valid but should return an empty array